### PR TITLE
Added settings to reduce tokensize for OpenId

### PR DESCRIPTION
**Pull request by MatthijsKrempel**
For hosting in cloud based environments header size matters. Currently the default implementation of the OpenId serializes a lot of information into the security ticket, resulting into large headersizes of the request. With these three extra options the administrator will be able to control what claims are included into the ticket, reducing the size of the values in the header. Additional information that a service would require could be retrieved via an additional request to the OpenId server.

### CODE REVIEW

**Comment by MatthijsKrempel** [comment]
@PinpointTownes can you have a look?

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdServerSettings.Edit.cshtml [hunk 1]
```diff
@@ -175,6 +177,28 @@
     <span asp-validation-for="AccessTokenFormat"></span>
 </fieldset>
 
+<fieldset class="form-group" asp-validation-class-for="IncludeLegacyMicrosoftClaimsInToken">
+    <div class="custom-control custom-checkbox">
+        <input type="checkbox" class="custom-control-input" asp-for="IncludeLegacyMicrosoftClaimsInToken">
+        <label class="custom-control-label" asp-for="IncludeLegacyMicrosoftClaimsInToken">@T["Include Legacy MicrosoftClaims in the Token."]</label>
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdServerSettings.Edit.cshtml]: Don't use upper case on each new word.
Example:
```
"When this option is enabled, a new refresh token is issued for each refresh token request (and the previous one is automatically revoked)."
```

Comment (MatthijsKrempel) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdServerSettings.Edit.cshtml]: Done

**Comment by PinpointTownes** [comment]
Sounds like a good start, but shouldn't we go further and introduce a mapping feature that allows listing all the claims we want in access and identity tokens, possibly with a different name?

I'm pretty sure @sebastienros and I chatted about that last week. He certainly has some input about that feature.

**Comment by MatthijsKrempel** [comment]
Could you draw me a picture of what you would like the end result to look like? I would be happy to help.
For now, our production clusters are struggeling with the large tokensizes in headers. Our ingress and kestrel are throwing errors, so this is sort of a 'quick fix'.

**Comment by MatthijsKrempel** [comment]
@PinpointTownes do you have any idea what the end result would look like?

**Comment by PinpointTownes** [comment]
It could be as simple as 2 textboxes and 2 checkboxes: one for the input claim name, one for the output claim name and the 2 checkboxes for the claim destination (access and identity token).

Or it could be more elaborate and use some scripting engine. I believe @MichaelPetrinolis worked on something similar to allow mapping external claims. @MichaelPetrinolis do you confirm? :smile:

**Comment by MichaelPetrinolis** [comment]
I confirm what @PinpointTownes says, you could use IScriptingManager to filter the claims that will be included in the ticket. Providing a default script and adequate documentation/examples, that would be an easy and powerful feature. Check PR #3722 for a screenshot and how mapping external claims to local roles is done.

1 checkbox and 1 textbox 😄

**Comment by MatthijsKrempel** [comment]
@PinpointTownes if I pinkyswear that I will work on this mapper, can we put this in, in the mean time?

**Comment by PinpointTownes** [comment]
That's a question for @sebastienros, as I'm not sure what the current roadmap looks like.

If we "temporarily" merge this change, we'll need to replace it quickly to make sure it's not part of a public release (otherwise, it will be hard to remove it in a future version).

**Comment by agriffard** [comment]
@MatthijsKrempel Please conflicts to solve also here.

**Comment by agriffard** [comment]
@MatthijsKrempel Please merge dev and solve the conflicts.

**Comment by Piedone** [comment]
I sincerely apologize for us taking so much time here. I checked out the PR, going over old ones, and this would be quite useful.

Is this something you'd like to revisit @MatthijsKrempel  or should we close? @kevinchalet with a review?

**Comment by MatthijsKrempel** [comment]
You can close, I moved on 🤣

________________________________

**Comment by Piedone** [comment]
Thanks for your quick reply! I've opened https://github.com/OrchardCMS/OrchardCore/issues/15067 to keep track of this, should anybody be interested to take over.

**Pull request closed without merging**